pkg/oceancd/model/strategy/strategy.go

```diff
@@ -0,0 +1,49 @@
+package strategy
+
+type RollingUpdateStrategy struct {
+	Steps []Step `json:"steps"`
+}
+
+type CanaryStrategy struct {
+	Steps []Step `json:"steps"`
+}
+
+func (c *CanaryStrategy) GetHeaderRouteMatchesBySteps() map[string][]Match {
+	matches := map[string][]Match{}
+
+	for _, step := range c.Steps {
+		matches[step.Name] = append(matches[step.Name], step.SetHeaderRoute.Match...)
+	}
+
+	return matches
+}
+
+func (r *RollingUpdateStrategy) GetHeaderRouteMatchesBySteps() map[string][]Match {
+	matches := map[string][]Match{}
+
+	for _, step := range r.Steps {
+		matches[step.Name] = append(matches[step.Name], step.SetHeaderRoute.Match...)
+	}
+
+	return matches
+}
+
+type Step struct {
+	Name           string         `json:"name"`
+	SetHeaderRoute SetHeaderRoute `json:"setHeaderRoute"`
+}
+
+type SetHeaderRoute struct {
```

Comment:
here you are missing the name: 
example:
` {
                    "name": "second",
                    "setHeaderRoute": {
                        "name": "set-header-2",
                        "match": [
                            {
                                "headerName": "beta-tester",
                                "headerValue": {
                                    "exact": "enabled"
                                }`

Reply:
@caduri I don't use the field, and previously we discussed that we don't need to implement unused fields. Please confirm